app/main.py

```diff
@@ -1,13 +1,18 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"erros": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": "failed" if errors is None else "passed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [format_single_linter_file(error) for error in linter_report]
```

Comment:
You should pass 2 arguments to `format_single_linter_file`
```suggestion
    return [format_single_linter_file(path, error) for path, error in linter_report.items()]
```